src/jit/unwind.cpp

```diff
@@ -199,6 +186,23 @@ void Compiler::unwindBegPrologCFI()
 
 void Compiler::unwindPushPopMaskCFI(regMaskTP regMask, bool isFloat)
 {
+#if defined(_TARGET_ARM_)
```

Comment:
It would be useful to add a comment here to explain why ARM walks the mask "backwards", whereas other platforms do not.

Reply:
LLVM reverses unwind opcodes that are fed to the streamer, but I for the life of me can't figure out why:
```
   // Copy the unwind opcodes
   for (size_t i = OpBegins.size() - 1; i > 0; --i)
     for (size_t j = OpBegins[i - 1], end = OpBegins[i]; j < end; ++j)
       OpStreamer.EmitByte(Ops[j]);
```
http://llvm.org/doxygen/ARMUnwindOpAsm_8cpp_source.html, lines 187-190

Reply:
I thought about it for a while and I believe the right way to do it is to leave this part as it was, walking the mask start to finish, but change CoreRT's ObjectWriter instead to collect a set of CFI codes sharing the same CodeOffset (generated from one push instruction pushing multiple registers) and generate one ARM unwind opcode for the whole set with a register mask as the argument. That way the fact that LLVM reverses unwind opcodes would only play for us, reversing the effect of the prolog, but leaving registers within the same `push` alone.